docs/DeveloperGuide.rst

```diff
@@ -0,0 +1,182 @@
+Developer Guide: StarCellBio
+============================
+
+The intent of this guide is to:
+
+    * Describe the Architecture of StarCellBio
+    * Create a Virtual Environment for developing StarCellBio
+    * Clone starcellbio_html
+    * Setup Development Environment
+    * Using watch.py to compile .soy and .gss files into .js and .css files
+    * Start the StarCellBio development server on localhost
+    * Use StarCellBio server on localhost
+    * Modify/Deploy/Use StarCellBio html_app
+    * Modify/Deploy/Use StarCellBio instructor
+    * Create/Modify/Deploy/Use StarCellBio AssignmentBuilder
+    * Create/Modify/Deploy/Use StarCellBio Lab Notebook
+
+
+Describe the Architecture of StarCellBio
+----------------------------------------
+
+    :download:`StarCellBio Architecture <StarCellBio.pdf>`
+
+Create a Virtual Environment for developing StarCellBio
+-------------------------------------------------------
+"A Virtual Environment ... is an isolated working copy of Python which allows
+you to work on a specific project without worry of affecting other project."
+
+Reference: `http://docs.python-guide.org/en/latest/dev/vertualenvs/ <http://docs.python-guide.org/en/latest/dev/vertualenvs/>`_
+
+This sample expects the ``PyVENV`` folder to contain the Virtual Environment for the sample.
+
+To ``CREATE`` the PyVENV Virtual Environment::
+
+    $ cd ~
+    $ pip install virtualenv
+    $ virtualenv PyVENV
+
+To ``ACTIVATE`` it::
+
+    $ source ~/PyVENV/bin/activate
+
+To ``DEACTIVATE`` it::
+
+    (PyVENV) $ deactivate
+
+To ``DESTROY`` it::
+
+    (PyVENV) $ deactivate
+    $ rm -Rv ~/PyVENV
+
+You may also want to install Virtualenvwrapper to help manage Virtual Environments although it is not required.
+Reference: 'https://vitualenvwrapper.readthedocs.org/ <https://vitualenvwrapper.readthedocs.org/>'
+
+Clone starcellbio_html
+----------------------
+You may also need to install Homebrew and pip as these are required.
+
+.. note:: This example uses assumes that PROJECT_HOME is ~/starcellbio/starcellbio_html.
+You will need to adjust PROJECT_HOME to the location of your starcellbio_html project.
+
+In a terminal window execute the following commands::
+
+    $ source ~/PyVENV/bin/activate
+    (PyVENV) $ git clone git://github.com/starteam/starcellbio_html
+
+
+Setup Development Environment
+-----------------------------
+
+Using PyCharm
+    View->Tool Windows->Database + Data Source -> MySQL
+    Install the database drivers if prompted
+
+
+mysqld command starts database
+
+Reference: `Installing MySQL on Mac OS X <https://rtcamp.com/tutorials/mac/osx-brew-php-mysql-nginx/>`_
+
+If you need to install MySQL on Mac OS X, execute the following command::
+
+    $ brew install mysql --enable-debugging
```

Comment:
I think we should consider adding a Vagrantfile to the repo for development.  That will make setting up the development environment easier and we can control the packages/not deal with instructions for a lot of different systems